From 87beff6c0d67f5ae8f088a49469bf5262d75023b Mon Sep 17 00:00:00 2001
From: Jiping Ma <jiping.ma2@windriver.com>
Date: Tue, 9 Apr 2024 07:30:05 +0000
Subject: [PATCH 04/10] bnxt_en: skb_frag_fill_page_desc() replace of
 __skb_frag_set_page

This patch accommodates commit 278fda0d52f6 ("net: remove __skb_frag_set_page()")
merged in the v6.5 development cycle, by applying a similar change to the
out-of-tree bnxt_en driver to resolve the compilation errors encountered with
the v6.6 kernel, which are listed below.

References:
* https://git.yoctoproject.org/linux-yocto/commit/?h=b51f4113ebb0
* https://git.yoctoproject.org/linux-yocto/commit/?h=278fda0d52f6

Resolved compilation errors:

bnxt_en-1.10.2-227.0.130.0/bnxt.c: In function __bnxt_rx_agg_pages:
bnxt_en-1.10.2-227.0.130.0/bnxt.c:1545:3: error: implicit declaration\
 of function __skb_frag_set_page; did you mean skb_frag_page? \
 [-Werror=implicit-function-declaration]
 1545 |   __skb_frag_set_page(frag, cons_rx_buf->page);
      |   ^~~~~~~~~~~~~~~~~~~

Signed-off-by: Jiping Ma <jiping.ma2@windriver.com>
---
 bnxt_en-1.10.2-227.0.130.0/bnxt.c | 10 +++-------
 1 file changed, 3 insertions(+), 7 deletions(-)

diff --git a/bnxt_en-1.10.2-227.0.130.0/bnxt.c b/bnxt_en-1.10.2-227.0.130.0/bnxt.c
index f464809..b763655 100644
--- a/bnxt_en-1.10.2-227.0.130.0/bnxt.c
+++ b/bnxt_en-1.10.2-227.0.130.0/bnxt.c
@@ -1540,9 +1540,8 @@ static u32 __bnxt_rx_agg_pages(struct bnxt *bp,
 			    RX_AGG_CMP_LEN) >> RX_AGG_CMP_LEN_SHIFT;
 
 		cons_rx_buf = &rxr->rx_agg_ring[cons];
-		skb_frag_off_set(frag, cons_rx_buf->offset);
-		skb_frag_size_set(frag, frag_len);
-		__skb_frag_set_page(frag, cons_rx_buf->page);
+		skb_frag_fill_page_desc(frag, cons_rx_buf->page,
+					cons_rx_buf->offset, frag_len);
 		shinfo->nr_frags = i + 1;
 		__clear_bit(cons, rxr->rx_agg_bmap);
 
@@ -1558,10 +1557,7 @@ static u32 __bnxt_rx_agg_pages(struct bnxt *bp,
 			xdp_buff_set_frag_pfmemalloc(xdp);
 
 		if (bnxt_alloc_rx_page(bp, rxr, prod, GFP_ATOMIC) != 0) {
-			unsigned int nr_frags;
-
-			nr_frags = --shinfo->nr_frags;
-			__skb_frag_set_page(&shinfo->frags[nr_frags], NULL);
+			--shinfo->nr_frags;
 			cons_rx_buf->page = page;
 
 			/* Update prod since possibly some pages have been
-- 
2.43.0

